src/listen-test.js

```diff
@@ -7,118 +7,98 @@ const dirtyChai = require('dirty-chai')
+      let finish
+      let done = new Promise((resolve) => {
+        finish = resolve
       })
 
-      listener.listen(addrs[0], () => {
-        const socket1 = transport.dial(addrs[0], () => {
-          listener.close(finish)
-        })
+      const listener = transport.createListener((conn) => pipe(conn, conn))
 
-        pull(
-          transport.dial(addrs[0]),
-          pull.onEnd(() => {
-            finish()
-          })
-        )
-
-        pull(
-          pull.values([Buffer.from('Some data that is never handled')]),
-          socket1,
-          pull.onEnd(() => {
-            finish()
-          })
-        )
+      // Listen
+      await listener.listen(addrs[0])
+
+      // Create two connections to the listener
+      const socket1 = await transport.dial(addrs[0])
+      await transport.dial(addrs[0])
+
+      pipe(
+        [Buffer.from('Some data that is never handled')],
+        socket1
+      ).then(() => {
+        finish()
       })
+
+      // Closer the listener (will take a couple of seconds to time out)
+      await listener.close()
+
+      // Pipe should have completed
+      await done
     })
 
     describe('events', () => {
-      // eslint-disable-next-line
-      // TODO: figure out why it fails in the full test suite
-      it.skip('connection', (done) => {
-        const finish = plan(2, done)
-
+      it('connection', (done) => {
         const listener = transport.createListener()
 
-        listener.on('connection', (conn) => {
+        listener.on('connection', async (conn) => {
           expect(conn).to.exist()
-          finish()
-        })
-
-        listener.listen(addrs[0], () => {
-          transport.dial(addrs[0], () => {
-            listener.close(finish)
-          })
-        })
+          listener.close()
+          done()
+        });
+
+        (async () => {
+          await listener.listen(addrs[0])
+          await transport.dial(addrs[0])
+        })()
       })
 
       it('listening', (done) => {
         const listener = transport.createListener()
-        listener.on('listening', () => {
-          listener.close(done)
+        listener.on('listening', async () => {
+          await listener.close()
+          done()
         })
         listener.listen(addrs[0])
       })
 
-      // eslint-disable-next-line
-      // TODO: how to get the listener to emit an error?
-      it.skip('error', (done) => {
+      it('error', (done) => {
         const listener = transport.createListener()
         listener.on('error', (err) => {
           expect(err).to.exist()
-          listener.close(done)
+          listener.close()
+          done()
         })
+        listener.emit('error', new Error('my err'))
       })
 
       it('close', (done) => {
-        const finish = plan(2, done)
         const listener = transport.createListener()
-        listener.on('close', finish)
+        listener.on('close', done);
```

Comment:
Unnecessary ;

Reply:
Somewhat weirdly, if there is no semi-colon, it sees the block below as being the parameter to a function call, ie
```Javascript
listener.on('close', done)(async () => {
          await listener.listen(addrs[0])
          await listener.close()
        })()
```